classes/local/accredible.php

```diff
@@ -0,0 +1,98 @@
+<?php
+// This file is part of the Accredible Certificate module for Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace mod_accredible\local;
+
+/**
+ * Defines local functions for handling interactions with the 'accredible' database table.
+ *
+ * @package    mod_accredible
+ * @subpackage accredible
+ * @copyright  Accredible <dev@accredible.com>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class accredible {
+
+    /**
+     * Saves or updates an Accredible plugin instance record in the 'accredible' table.
+     * This function handles both the creation of new records and the updating of existing ones.
+     *
+     * @param stdClass $post An object containing the incoming data from the form submission.
+     * @param stdClass|null $accrediblecertificate Optional. Existing certificate data to be updated.
+     * @param int|null $groupid Optional. The group ID associated with the plugin instance.
```

Comment:
Question - Why does `$groupid` need to be supported as the third arg? I was wondering if the function can set the right `$groupid` from `$post` so that the thrid arg would be redundant.

Reply:
Refactored to avoid passing groupid